lib/inspec/control_eval_context.rb

```diff
@@ -117,10 +117,21 @@ def to_s
         end
 
         define_method :register_control do |control, &block|
-          if @skip_file || !profile_context_owner.profile_supports_os?
+          if @skip_file
             ::Inspec::Rule.set_skip_rule(control, true)
           end
 
+          unless profile_context_owner.profile_supports_platform?
+            platform = inspec.platform
+            msg = "Platform #{platform.name}/#{platform.release} is not supported with profile '#{profile_context_owner.profile_id}'"
```

Comment:
"supported with" feels like it might not be the right English to use. I'm wondering if this might be better:

`"Profile #{profile_context_owner.profile_id} is not supported on platform #{platform.name}/#{platform.release}."`